File: analysis/import/SCMLogParser/src/main/java/de/maibornwolff/codecharta/importer/scmlogparser/input/metrics/NumberOfAuthors.java
package de.maibornwolff.codecharta.importer.scmlogparser.input.metrics;

import de.maibornwolff.codecharta.importer.scmlogparser.input.Commit;

import java.util.HashSet;
import java.util.Set;

public class NumberOfAuthors implements Metric {
    private final Set<String> authors = new HashSet<>();

    @Override
    public String description() {
        return "Number of Authors: Number of authors that contributed to this file.";
    }

    @Override
    public String metricName() {
        return "number_of_authors";
    }

    @Override
    public void registerCommit(Commit commit) {
        authors.add(commit.getAuthor());
    }

    @Override
    public Number value() {
        return authors.size();
    }
}

File: analysis/import/SCMLogParser/src/main/kotlin/de/maibornwolff/codecharta/importer/scmlogparser/input/metrics/NumberOfAuthors.kt
package de.maibornwolff.codecharta.importer.scmlogparser.input.metrics

import de.maibornwolff.codecharta.importer.scmlogparser.input.Commit

class NumberOfAuthors: Metric {
    private val authors = mutableSetOf<String>()

    override fun description(): String {
        return "Number of Authors: Number of authors that contributed to this file."
    }

    override fun metricName(): String {
        return "number_of_authors"
    }

    override fun registerCommit(commit: Commit) {
        authors.add(commit.author)
    }

    override fun value(): Number {
        return authors.size
    }
}
